### PR TITLE
kube-{apiserver,ctrl-mgr}: unify into DeprecatedInsecureServingOptions

### PR DESCRIPTION
**Pull request by sttts**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/kube-scheduler/app/options/insecure_serving.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (o *CombinedInsecureServingOptions) ApplyToFromLoadedConfig(c *schedulerapp
 	return o.applyTo(c, componentConfig)
 }
 
-func updateAddressFromInsecureServingOptions(addr *string, is *controlleroptions.InsecureServingOptions) error {
+func updateAddressFromInsecureServingOptions(addr *string, is *apiserveroptions.DeprecatedInsecureServingOptions) error {
```
Comment (stewart-yu) [inline cmd/kube-scheduler/app/options/insecure_serving.go]: Should we rename `updateAddressFromInsecureServingOptions` to `DeprecatedInsecureServingOptions`?

File: cmd/kube-scheduler/app/options/insecure_serving.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func updateAddressFromInsecureServingOptions(addr *string, is *controlleroptions
 	return nil
 }
 
-func updateInsecureServingOptionsFromAddress(is *controlleroptions.InsecureServingOptions, addr string) error {
+func updateInsecureServingOptionsFromAddress(is *apiserveroptions.DeprecatedInsecureServingOptions, addr string) error {
 	if is == nil {
```
Comment (stewart-yu) [inline cmd/kube-scheduler/app/options/insecure_serving.go]: If possible, rename this function?

File: cmd/kube-scheduler/app/options/insecure_serving.go (hunk 1 above)
Comment (sttts) [reply cmd/kube-scheduler/app/options/insecure_serving.go]: done

File: cmd/kube-scheduler/app/options/insecure_serving.go (hunk 2 above)
Comment (sttts) [reply cmd/kube-scheduler/app/options/insecure_serving.go]: done

**Comment by sttts** [comment]
> staging/src/k8s.io/apiserver/pkg/server/options/serving_test.go:477 compile error

Fixed

**Comment by sttts** [comment]
/assign @andrewsykim @deads2k @k82cn 

for approval for the ripples into kube-scheduler, kube-controller-manger and cloud-controller-manager due to the moves package.

File: cmd/kube-apiserver/app/options/options.go [hunk 3]
```diff
@@ -128,7 +128,7 @@ func (s *ServerRunOptions) Flags() (fss apiserverflag.NamedFlagSets) {
 	s.Etcd.AddFlags(fss.FlagSet("etcd"))
 	s.SecureServing.AddFlags(fss.FlagSet("secure serving"))
 	s.InsecureServing.AddFlags(fss.FlagSet("insecure serving"))
-	s.InsecureServing.AddDeprecatedFlags(fss.FlagSet("insecure serving"))
+	s.InsecureServing.AddUnqualifiedFlags(fss.FlagSet("insecure serving")) // TODO: remove it until kops stops using `--address`
```
Comment (deads2k) [inline cmd/kube-apiserver/app/options/options.go]: I don't think we'll actually wait for kops.  When we're past the deprecation cycle time, we can remove it.

**Comment by deads2k** [comment]
Alright, I see the reason for doing this.  It's clearly named as Deprecated.  I can live with the move to the genericapiserver.

/lgtm

**Comment by sttts** [comment]
@k82cn @luxas @aveshagarwal approved?

File: cmd/kube-apiserver/app/options/options.go (hunk 3 above)
Comment (sttts) [reply cmd/kube-apiserver/app/options/options.go]: We should formally deprecate insecure ports for all components in 1.12 and remove them all at once in 1.13.

Comment (luxas) [reply cmd/kube-apiserver/app/options/options.go]: > We should formally deprecate insecure ports for all components in 1.12 and remove them all at once in 1.13.

Yes, I agree. Let's make that super clear in the release notes.

File: staging/src/k8s.io/apiserver/pkg/server/options/deprecated_insecure_serving.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+// AddUnqualifiedFlags adds flags related to insecure serving without the --insecure prefix to the specified FlagSet.
+func (s *DeprecatedInsecureServingOptions) AddUnqualifiedFlags(fs *pflag.FlagSet) {
+	if s == nil {
+		return
+	}
+
+	fs.IPVar(&s.BindAddress, "address", s.BindAddress,
+		"DEPRECATED: see --bind-address instead.")
```
Comment (luxas) [inline staging/src/k8s.io/apiserver/pkg/server/options/deprecated_insecure_serving.go]: Can we get rid of these in v1.13?

**Review by luxas — APPROVED** [review]
This looks like a very nice change @sttts, thanks for working on unifying all of this stuff. After this I guess we should do something wrt unifying the healthz and metrics serving options ;)

/approve
I think this direction is great, but somebody else can do the full line-by-line review and give lgtm

**Review by bsalamat — COMMENTED** [review]
/approve 

for the scheduler changes. Please address the failures.

File: cmd/kube-controller-manager/app/options/options.go [hunk 5]
```diff
@@ -341,10 +341,10 @@ func (s *KubeControllerManagerOptions) ApplyTo(c *kubecontrollerconfig.Config) e
 	if err := s.ServiceController.ApplyTo(&c.ComponentConfig.ServiceController); err != nil {
 		return err
 	}
-	if err := s.SecureServing.ApplyTo(&c.SecureServing); err != nil {
+	if err := s.InsecureServing.ApplyTo(&c.InsecureServing); err != nil {
 		return err
 	}
-	if err := s.InsecureServing.ApplyTo(&c.InsecureServing); err != nil {
+	if err := s.SecureServing.ApplyTo(&c.SecureServing); err != nil {
```
Comment (roycaihw) [inline cmd/kube-controller-manager/app/options/options.go]: nit: why do we change the order of L344 and L347?

Comment (sttts) [reply cmd/kube-controller-manager/app/options/options.go]: Like in kube-apiserver we prefer a loopback connection through insecure serving. Not really super important, but more uniform.

**Pull request merged**